packages/client/hmi-client/src/components/model/tera-model-variables.vue

```diff
@@ -1,19 +1,22 @@
 <template>
 	<component
-		:is="tables"
+		:is="variablesComponent"
 		:model="transientModel"
 		:mmt="mmt"
 		:mmt-params="mmtParams"
 		:readonly="readonly"
 		@update-model="$emit('update-model', $event)"
-		@update-state="onUpdateState"
-		@update-parameter="onUpdateParameter"
+		@update-state="(e: any) => onUpdate('state', e)"
+		@update-parameter="(e: any) => onUpdate('parameter', e)"
+		@update-observable="(e: any) => onUpdate('observable', e)"
+		@update-transition="(e: any) => onUpdate('transition', e)"
+		@update-time="(e: any) => onUpdate('time', e)"
 	/>
 </template>
```

Comment:
I am not crazy about the naming here, between
- tera-model
- tera-model-variables
- tera-variables
- tera-variable-entry
- tera-parameters

And that the terms variables and parameters are kind of exchangeable....

I find it hard to keep track of what is what, can you brief on the roles they play? Let's see if we can have a better naming scheme.

Reply:
In my mind a variable counts as a state, transition, parameter, observable. Would there be a better name to generally describe these?

Reply:
I would describe these (state, transition ... etc) as components or parts of a petrinet-model ... or alternatively components of a compartmental-model in a hazy kind of way. 

Specifically:
- "transition" describe relationships between multiple "state"
- "state" points to a "parameter"
- "observable" contains "state" or "parameter" with some relationship bindings
- "parameter" are values or distributions

Reply:
I think `model-parts` sounds good - there a modelparts folder in the server. `model-components` could mixed up with a vue component.